google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/ReadOnlyStalenessUtil.java

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spanner.jdbc;
+
+import com.google.api.client.util.DateTime;
+import com.google.cloud.Timestamp;
+import com.google.cloud.spanner.ErrorCode;
+import com.google.cloud.spanner.SpannerException;
+import com.google.cloud.spanner.SpannerExceptionFactory;
+import com.google.cloud.spanner.TimestampBound;
+import com.google.cloud.spanner.TimestampBound.Mode;
+import com.google.protobuf.Duration;
+import com.google.protobuf.util.Durations;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Util class for parsing and converting ReadOnlyStaleness values to/from strings. This util is used
+ * to parse client side statements and values for read only staleness for read-only transactions on
+ * Cloud Spanner.
+ */
+class ReadOnlyStalenessUtil {
+  /**
+   * Parses an RFC3339 date/time value with millisecond precision and returns this as a {@link
+   * Timestamp}. Although {@link Timestamp} has nanosecond precision, this method is only used for
+   * parsing read-only staleness values, and more than millisecond precision is not needed for that.
```

Comment:
minor can address in upcoming - I would just replace this with a todo for nanosecond precision